libs/kotaemon/kotaemon/indices/vectorindex.py

```diff
@@ -78,11 +78,27 @@ def run(self, text: str | list[str] | Document | list[Document]):
         if self.cache_dir:
             file_name = Path(input_[0].metadata["file_name"])
             for i in range(len(input_)):
-                print(Path(self.cache_dir) / f"{file_name.stem}_{self.count_+i}.md")
+                markdown_content = ""
+                if "page_label" in input_[i].metadata:
+                    page_label = str(input_[i].metadata["page_label"])
+                    markdown_content += f"Page label: {page_label}"
+                if "file_name" in input_[i].metadata:
+                    filename = input_[i].metadata["file_name"]
+                    markdown_content += f"\nFile name: {filename}"
+                if "section" in input_[i].metadata:
+                    section = input_[i].metadata["section"]
+                    markdown_content += f"\nSection: {section}"
+                if "type" in input_[i].metadata:
+                    if input_[i].metadata["type"] == "image":
+                        image_origin = input_[i].metadata["image_origin"]
+                        markdown_content += f"\nImage origin: {image_origin}"
```

Comment:
The image_origin is a base64 string. You can try adding the base64 image in markdown file like [this](https://stackoverflow.com/questions/50817518/hard-code-markdown-images).

After that, you can try with Markdown preview tool like https://markdownlivepreview.com/

If the Markdown couldn't show image, then I suggest convert the image from base64 string to PNG file.